javalib/src/main/scala/java/util/AbstractCollection.scala

```diff
@@ -0,0 +1,89 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.immutable.{ List => SList }
+
+abstract class AbstractCollection[E]() extends java.util.Collection[E] {
+  def iterator(): Iterator[E]
+  def size(): Int
+
+  @tailrec
+  private def iterateAndDo[X](i: Iterator[E], f: (E) => X, part: SList[X] = SList()): SList[X] =
```

Comment:
Unless it hurts readability, stay under 80 characters. Here you can easily break before `part`, indent 4 spaces. Then add `{}` around the body of the method, because the signature spans more than 1 line.